bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/InstallationCleaner.java

```diff
@@ -104,6 +93,23 @@ void cleanup() throws IOException {
         }
     }
 
+    // In case of timeout, we attempt to do a cleanup only if the cleanupMarker exists
+    synchronized void cleanupTimeout() throws IOException, InterruptedException {
+        if (Files.notExists(cleanupMarker)) {
+            return;
+        }
+        // In case we have a running process, kill it to avoid it to continue any cleanup.
+        if (newProcess && process != null) {
+            process.destroyForcibly();
+            process.waitFor(environment.getTimeout(), TimeUnit.SECONDS);
+            process = null;
+        }
+        // Do a last cleanup, in case the cleanupMarker still exists (could have been deleted by running process).
+        if (Files.exists(cleanupMarker)) {
+            cleanup();
```

Comment:
This could potentially launch another process. We should probably just invoke the `deleteDirectory()` at this point.

Reply:
Yes, that is done in purpose. On Windows we need the the external process. The cleanup waits until the process terminate (with a timeout).

Reply:
That is in case, the previous process didn't complete the deletion for some reason (timeout and process forcibly terminated from the caller thread), we start a new process to finalize.